llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -3275,8 +3307,9 @@ class BoUpSLP {
                           SmallVectorImpl<Value *> *AltScalars = nullptr) const;
 
     /// Return true if this is a non-power-of-2 node.
-    bool isNonPowOf2Vec() const {
```

Comment:
Request: Please leave the original method, and add a new one called isAllowedNonPowOfTwo.